### DISCUSSION TITLE
HttpClient Dns Changes

### DISCUSSION THREAD

**Discussion by MartinMogh** [dotnet/runtime#72514] [Q&A] [answered]
I have been using this code to avoid dns issues.

```csharp
  using var socketHandler = new SocketsHttpHandler
  {
      PooledConnectionIdleTimeout = TimeSpan.FromMinutes(1),
      PooledConnectionLifetime = TimeSpan.FromMinutes(5),
  };

  using var httpClient = new HttpClient(socketHandler);

```

But i am worried that using PooledConnectionLifetime will cancel download of large files.

Is there anyway of avoiding dns issus without PooledConnectionLifetime?

**Comment by huoyaoyuan** [comment]
You can ultimately do the DNS yourself like https://github.com/dotnet/runtime/issues/45144#issuecomment-766977127 . This is somehow overkill though.

**Reply by MartinMogh** [reply]
I was hoping for a simpler solution

**Comment by stephentoub** [comment, marked as answer]
> But i am worried that using PooledConnectionLifetime will cancel download of large files.

It won't. This setting doesn't affect connections actively being used. It determines whether to get rid of idle connections that are too old.

**Reply by MartinMogh** [reply]
In the microsoft docs it says: "This property defines maximal connection lifetime in the pool regardless of whether the connection is idle or active."

**Reply by stephentoub** [reply]
I realize that's confusing wording, but it means it's counting all of the time the connection has been alive, regardless of whether that time was spent idle or active.  It will only tear down a connection that's idle.

**Reply by stephentoub** [reply]
(I opened https://github.com/dotnet/dotnet-api-docs/pull/8241 to clarify.)

**Reply by MartinMogh** [reply]
thank you
